### DISCUSSION TITLE
Porosity Calculation during the simulation

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#23014] [Q&A Modules: General]
Hello MOOSE experts,

I have a porous microstructure which is expanding and contracting during the simulation.
I want to track its average porosity change. Can you please suggest me an example to do so?
I shall be very thankful.

Best,
Avtar

**Comment by GiudGiud** [comment]
Hello

So you already have the simulation and just need to measure the change?
What is porosity in your simulation? A material property? An auxvariable? Or is it not computed currently

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello Guillaume,

I have porous microstructure (Solid material with voids). Currently, I am not computing the porosity parameter.
Basically, I want to track the change in porosity upon swelling and contraction process that I am doing through lithiation and delithiation phenomena.

Any suggestions?

Best,
Avtar

**Reply by GiudGiud** [reply]
@jiangwen84 @sapitts on modeling a porous microstructure

**Reply by avtarsinghh1991** [reply]
Thanks,

I could do it through total volume expansion of mesh elements which I can calculate through volume postprocessor and then substracting from the total volume bound as I am working on cube. But the cube distorts a lot during the swelling and contraction makes it difficult to work through that route. 

Best,
Avtar
